components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/EndpointUtil.java

```diff
@@ -922,58 +927,83 @@ private static List<String> getOIDCScopeNames() throws IdentityOAuthAdminExcepti
     }
 
     /**
-     * Drop OIDC and unregistered scopes from consent required scopes.
+     * Return a list of consent requested OIDC scopes
      *
      * @param params OAuth2 parameters.
-     * @return consent required scopes
+     * @return consent requested OIDC scopes in lower case
      * @throws OAuthSystemException If retrieving OIDC scopes failed.
      */
-    private static List<String> dropOIDCAndUnregisteredScopesFromConsentRequiredScopes(OAuth2Parameters params)
+    private static List<String> getRequestedOIDCScopes(OAuth2Parameters params)
+            throws OAuthSystemException {
+
+        Set<String> allowedScopes = params.getScopes();
+        List<String> requestedOIDCScopes = new ArrayList<>();
+        try {
+            startTenantFlow(params.getTenantDomain());
+
+            // Get registered OIDC scopes.
+            String[] oidcScopes = oAuthAdminService.getScopeNames();
```

Comment:
We should have a method to get the same passing tenantDomain now

Reply:
What you have done is perfectly correct. But would be good to use the method that gets the tenant id or domain explicitly and remove starting tenant flows to avoid error prone cases in future modifications

Reply:
https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/bd890b3f900294f72c9ba50e60e53caa35c7382b/components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java#L1875